### ISSUE TITLE
add support for forward_params in TextClassificationPipeline

### ISSUE THREAD

**Issue by asai-carbon** [huggingface/transformers#29271]
### Feature request

I want to input a global attention mask tensor (to be used as part of `LongformerForSequenceClassification`) as part of the `__call__` function for `TextClassificationPipeline`. It has to be input into the `call` function as it is [used at run-time](https://huggingface.co/docs/transformers/v4.38.1/en/model_doc/longformer#longformer-self-attention). I am using `TextClassificationPipeline` instead of the direct model object `LongformerForSequenceClassification` because the pipeline is the default for inference on SageMaker endpoints. 

However, the `_sanitize_parameters` method for `TextClassificationPipeline` [does not output](https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/text_classification.py#L119) `forward_params` which would be used as input into the `forward` method that performs inference by calling the model. It actually [assigns all additional keyword arguments](https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/text_classification.py#L94) for the function to the `preprocess_params` variable.

 I would like the model to be called with an additional argument of the global attention mask. In order to do this, the `TextClassificationPipeline._sanitize_parameters` method could be refactored to accept forward parameters arguments like global attention mask tensor, which would help global attention mask to be applied at inference time.

If however, there is an alternative to the proposed feature, like making a custom inference script to handle inference with the global attention mask as an explicit parameter to the model directly as opposed to the pipeline object as mentioned [here](https://huggingface.co/docs/sagemaker/en/inference#user-defined-code-and-modules), I would like to understand better how to do that as well.

### Motivation

I am currently attempting to improve the performance of a `LongformerForSequenceClassification` model I have fine-tuned by changing the current global attention configuration, which currently applies global attention to only the separator token. I would like the ability to be able to pass in a global attention mask tensor at inference time and have it be processed by a Huggingface pipeline object, specifically the `TextClassificationPipeline` object.

### Your contribution

If those in the community who respond to this issue feel a PR is necessary and no existing solution exists for this issue, I would be happy to help by submitting a PR.

**Comment by ArthurZucker** [comment]
FYI @Rocketknight1

**Comment by Rocketknight1** [comment]
@philschmid this seems more like a Sagemaker thing than a pipeline issue - do you have any scripts here? If not I can work on modifying `TextClassificationPipeline` to accept arbitrary model arguments.

**Comment by philschmid** [comment]
Yes, @asai-carbon you can provide a custom `inference.py` with the modifications you need to make.
